src/spatialdata/_core/spatialdata.py

```diff
@@ -184,13 +229,167 @@ def from_elements_dict(elements_dict: dict[str, SpatialElement | AnnData]) -> Sp
+
+        Parameters
+        ----------
+        table
+            The AnnData object containing the data table.
+        region
+            The name of the target element for the table annotation.
+        region_key
+            The key for the region annotation column in `table.obs`.
+        instance_key
+            The key for the instance annotation column in `table.obs`.
+
+        Raises
+        ------
+        ValueError
+            If `region_key` is not present in the `table.obs` columns.
+        ValueError
+            If `instance_key` is not present in the `table.obs` columns.
+        """
+        if region_key not in table.obs:
+            raise ValueError(f"Specified region_key, {region_key}, not in table.obs")
+        if instance_key not in table.obs:
+            raise ValueError(f"Specified instance_key, {instance_key}, not in table.obs")
+        attrs = {
+            TableModel.REGION_KEY: region,
+            TableModel.REGION_KEY_KEY: region_key,
+            TableModel.INSTANCE_KEY: instance_key,
+        }
+        check_target_region_column_symmetry(table, region_key, region)
+        table.uns[TableModel.ATTRS_KEY] = attrs
+
+    @staticmethod
+    def _change_table_annotation_target(
+        table: AnnData,
+        region: str | pd.Series,
+        region_key: None | str = None,
+        instance_key: None | str = None,
+    ) -> None:
+        """Change the annotation target of a table currently having annotation metadata already.
+
+        Parameters
+        ----------
+        table
+            The table already annotating a SpatialElement.
+        region
+            The name of the target SpatialElement for which the table annotation will be changed.
+        region_key
+            The name of the region key column in the table. If not provided, it will be extracted from the table's uns
+            attribute.
+        instance_key
+            The name of the instance key column in the table. If not provided, it will be extracted from the table's uns
+            attribute.
+
+        Raises
+        ------
+        ValueError
+            If no region_key is provided, and it is not present in both table.uns['spatialdata_attrs'] and table.obs.
+        ValueError
+            If provided region_key is not present in table.obs.
+        """
+        attrs = table.uns[TableModel.ATTRS_KEY]
+        table_region_key = attrs.get(TableModel.REGION_KEY_KEY)
+        if not region_key:
+            if not table_region_key:
+                raise ValueError(
+                    "No region_key in table.uns and no region_key provided as argument. Please specify " "region_key."
+                )
+            if table.obs.get(attrs[TableModel.REGION_KEY_KEY]) is None:
+                raise ValueError(
+                    f"Specified region_key in table.uns '{table_region_key}' is not "
+                    f"present as column in table.obs. Please specify region_key."
+                )
```

Comment:
Feels a bit confusing that the equivalent of this code for instance_key is in _validate_set_instance_key(). I would create such function also for region_key().

Also _validate_set_instance_key() is called only from here, not from model, so I would move it out of the model and put it in this file.

Reply:
resolved with [e6a3893](https://github.com/scverse/spatialdata/pull/410/commits/e6a3893d97ecaf7c12f1facf278400d370890499) although I did keep it in table model in case of in the future being called from elsewhere.